### PR TITLE
Add Function to Allow AWS Paths

### PR DESCRIPTION
**Pull request by wgranger**
This PR adds a computeURL function to allow a subject to be cropped if the URL is from out AWS bucket. 

@camallen

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -70,6 +70,17 @@ function cropImage(paramSource, res, next){
   });
 };
 
+function computeURL(static_host, u){
```
Comment (camallen) [inline server.js]: This is mostly there, i think having the ability to not just be a blanket s3.amazonaws.com would be good. I tried this out locally and it worked.

``` javascript
function escapeRegExp(string){
  return string.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, "\\$1");
}

function validUrl(arr, url) {
  return arr.some(function(arrVal) {
    var host_regex = new RegExp(escapeRegExp(arrVal), 'g');
    return url.match(host_regex);
  });
}

function computeURL(static_host, u){
  var whitelists = ['https://s3.amazonaws.com/astronomy-rewind'];
  if (validUrl(whitelists, u)) {
    return u
  } else {
    return static_host + u;
  }
};
```

**Comment by camallen** [comment]
can you make sure that all http URL's are https as well ?

**Comment by wgranger** [comment]
Yes, good call on the https. Need to remember to do that. I also tweaked the whitelist a bit because those urls are coming in from the WorldWide Telescope project without the prepended https.

**Comment by camallen** [comment]
Its slightly out of scope but can you update the static host default protocol to HTTPS too?

**Comment by camallen** [comment]
LGTM pending the final HTTPS config

**Comment by wgranger** [comment]
gotcha

**Comment by camallen** [comment]
LGTM

**Comment by wgranger** [comment]
Nice. Should I get another pair of eyes on this, or do you think we're all set?

**Comment by camallen** [comment]
If you've tested locally both your paths and zoo static paths then we're set, just make sure it still works for the old zoo static paths.

http://localhost:3000/crop?w=150&h=150&x=300&y=175&u=panoptes-uploads.zooniverse.org/production/subject_location/90a3b642-55e2-4583-a4fb-2f0abeb5b285.jpeg

http://localhost:8080/crop?w=150&h=150&x=300&y=175&u=s3.amazonaws.com/astronomy-rewind/AJ/0114/0001718.000.jpg

**Comment by wgranger** [comment]
Those all look fine locally. Are you able to merge?

**Comment by camallen** [comment]
@wgranger you should have collab rights now...go for it.

**Pull request merged**